src/Microsoft/Vkd3dCompiler/lib.c

```diff
@@ -132,6 +195,17 @@ extern int32_t D3DCompile2(const void *data, void* data_size, const char *filena
         option->value = true;
     }
 
+    if (flags & (D3DCOMPILE_PACK_MATRIX_ROW_MAJOR | D3DCOMPILE_PACK_MATRIX_COLUMN_MAJOR))
```

Comment:
Where does the code in this file actually come from?

Reply:
Comes from this file

https://github.com/Beyley/vkd3d/blob/9c2d740607a3234a5a98c3f084c2326569f6e183/libs/vkd3d-utils/vkd3d_utils_main.c#L265

All i have done is strip out the irrelevant parts

Reply:
Feels like a bit too much human intervention (and code that isn't ours, in our repo, effectively untracked)

I know it's cursed, but we could do something like
```c
#include "vkd3d_utils_main.c"
#undef D3D12GetDebugInterface
#undef D3D12CreateDeviceVKD3D
#undef D3D12CreateRootSignatureDeserializer
#undef D3D12CreateVersionedRootSignatureDeserializer
// etc...
```

Alternatively we could leave the irrelevant parts in... just not really keen on us having our own "buffer" of upstream

Reply:
> Feels like a bit too much human intervention (and code that isn't ours, in our repo, effectively untracked)
> 
> I know it's cursed, but we could do something like
> 
> ```c
> #include "vkd3d_utils_main.c"
> #undef D3D12GetDebugInterface
> #undef D3D12CreateDeviceVKD3D
> #undef D3D12CreateRootSignatureDeserializer
> #undef D3D12CreateVersionedRootSignatureDeserializer
> // etc...
> ```
> 
> Alternatively we could leave the irrelevant parts in... just not really keen on us having our own "buffer" of upstream

The irrelevant parts more meant it complicated the build process, since stuff like the logging macros were causing a billion undefined symbol errors i was not too keen on debugging/fixing, but i can give using the original source a shot with some well placed `def` and `undef`  if desired

Reply:
Yeah, the main thing is I had no idea this existed (yes I know clearly I don't pay that much attention) and wouldn't have thought to check here when updating vkd3d, and the same will definitely be said for whoever comes after you & I.

Eliminating as much human error as possible should be key, and this I feel like is definitely human-error-prone

Reply:
> Yeah, the main thing is I had no idea this existed (yes I know clearly I don't pay that much attention) and wouldn't have thought to check here when updating vkd3d, and the same will definitely be said for whoever comes after you & I.
> 
> Eliminating as much human error as possible should be key, and this I feel like is definitely human-error-prone

to be fair i *also* had to reference my own commits to figure out what i had done, so its a bit error prone yeah, i'll try to get the original source working at some point soon (and also update past `.405` so we get them LLVM 17 benefits)

Reply:
Notably 4 months ago they removed ms_abi so we are likely to have fixed #1634

Reply:
...although you probably already knew that as you've already linked the issue 🤦